### ISSUE TITLE
Instructions for integration into existing app

### ISSUE THREAD

**Issue by djMax** [forcedotcom/SalesforceMobileSDK-ReactNative#314]
The [forcereact](https://www.npmjs.com/package/forcereact) cli provides an easy way to get started if you are generating a new app. It contains some best practices that can be used to figure out how to integrate the SF RN sdk and the native sdk's, but it would be even better if a clearly documented step by step approach would be published about how to integrate the sdk's into an existing React Native app.

**Comment by bbirman** [comment]
Thanks for the feedback, we'll look into it

**Comment by oktalk** [comment]
Has anybody successfully done this? I would like to use a RN framework (like ignite) so it would be nice to have some guidance on integrating forcereact into an existing project.

**Comment by JFoxUK** [comment]
@djMax - did you ever manage this?

**Comment by JFoxUK** [comment]
@wmathurin any update here? @bbirman ?

**Comment by JFoxUK** [comment]
@wmathurin @bbirman any updates?

**Comment by JFoxUK** [comment]
@wmathurin @bbirman  - any update?

**Comment by galensalesforce** [comment]
Hi @JFoxUK  -- I'm a new PM on the Mobile SDK team. Thanks for following up on this. I just added this documentation update to our backlog with the target completion date being November 1. I'll follow up if anything changes.

**Comment by JFoxUK** [comment]
@galensalesforce  - any news?

**Comment by vishnudastk2005** [comment]
@galensalesforce . Hi galen, any updates.

**Comment by JFoxUK** [comment]
@galensalesforce its really disappointing that this keeps been ignored.

**Comment by galensalesforce** [comment]
Thanks all for following up about this -- and we appreciate your patience! MSDK is a small, open-source team that supports multiple platforms.

More info on this topic [here](https://trailhead.salesforce.com/trailblazer-community/feed/0D54V00007MYqaySAD). To summarize, we've decided that the best first step is to upgrade Mobile SDK support for React Native so that our forcereact command runs the RN CLI internally. This way, there will be far fewer steps required to add MSDK into an existing RN app to begin with. Our team is currently working on this upgrade for our 12.0 release.

**Comment by RohitSyngenta** [comment]
Is there anyway to handle the progress percentage with this SDK ?

**Comment by wmathurin** [comment]
@RohitSyngenta What did you mean by "Is there anyway to handle the progress percentage with this SDK ?"

**Comment by ivanoikon** [comment]
16 months and still no updates? It's absolutely annoying that this SDK doesn't have a way to integrate into an existing RN project.

**Comment by JFoxUK** [comment]
Yep, @galensalesforce promised to keep us in the loop 8m ago and never did.

**Comment by kqisalesforce** [comment]
Hi @JFoxUK, apologies for the delay. Galen has rotated onto a new team and is no longer on Mobile SDK. Here’s an update that was also posted on the [Trailblazer Community](https://trailhead.salesforce.com/trailblazer-community/feed/0D54V00007cmAsRSAU).

> The main reason we cannot provide step-by-step documentation for Mobile SDK integration into an existing RN app is our deep level of integration with the consuming app. We expect the generated app structure to allow these deeper integrations like biometric auth to work out of the box. I will make sure the team responds to any specific integration questions or issues that come up when trying to integrate Mobile SDK into your existing RN app. One thing we recommend is generating a sample app with [forcereact](https://www.npmjs.com/package/forcereact) then moving the assets into your existing app. One thing we have discussed is having [forcereact](https://www.npmjs.com/package/forcereact) leverage RN's CLI vs our own. This would have the generated application structure more closely align to RN CLI structure which could help with integrations into existing apps. 
> 
> I'm curious what type of app are you integrating or how are you wanting to use the Mobile SDK in your RN app. This could help us better understand how to help and possibly evolve the Mobile SDK. 
> 
> Again, I apologize if we caused you frustration our team is very small and we try to address every issue and question that comes in from our over 3000+ bespoke application teams.

**Comment by wmathurin** [comment]
With SDK 14.0 (React Native 0.82+), integration into an existing React Native app is significantly simpler than it was when this issue was opened. Here's what's changed and what's required.

**What changed:**

- **New Architecture is now mandatory in RN 0.82+** — there's no longer a legacy vs. bridgeless fork in the integration steps. If your app is on RN 0.82+, you're on the same architecture the SDK targets.
- **Android bridge is now in this repo** and uses React Native autolinking — no manual module registration needed. Adding `react-native-force` to your `package.json` and running `./gradlew` is sufficient.
- **`SalesforceReactActivityDelegate.loadApp()` override was removed** in SDK 14.0 — a previous source of black-screen issues in custom app setups.
- **The iOS AppDelegate pattern is now standard RN 0.82+ style** (`RCTReactNativeFactory` + `RCTReactNativeFactoryDelegate`), so if your app was generated with the RN CLI, the diff to add SDK support is small.

**Minimal integration steps:**

*Prerequisites: React Native 0.82+, SDK 14.0*

**1. Add the dependency**
```json
"react-native-force": "git+https://github.com/forcedotcom/SalesforceMobileSDK-ReactNative.git#dev"
```

**2. Add `bootconfig.json`** (Connected App credentials) to `android/app/src/main/assets/` and to your iOS app bundle:
```json
{ "remoteAccessConsumerKey": "<your-consumer-key>", "oauthRedirectURI": "<your-callback-uri>", "oauthScopes": ["api"], "isLocal": true, "startPage": "index.html", "errorPage": "error.html", "shouldAuthenticate": true, "attemptOfflineLoad": false }
```

**3. iOS — `AppDelegate.swift`**

Add `import SalesforceReact` and `import SalesforceSDKCore`. In `init()`, before anything else:
```swift
override init() {
    super.init()
    SalesforceReactSDKManager.initializeSDK()
}
```
Then wrap your React Native startup with the auth check:
```swift
AuthHelper.loginIfRequired {
    factory.startReactNative(withModuleName: "YourApp", in: self.window, launchOptions: launchOptions)
}
```
Add `SalesforceReact` to your `Podfile` (or use `use_mobile_sdk!` if you want all SDK libraries), then `pod install`.

**4. Android — `MainApplication.kt`**

Replace `SalesforceSDKManager.initNative(...)` (if present) with:
```kotlin
SalesforceReactSDKManager.initReactNative(applicationContext, MainActivity::class.java)
```
Autolinking handles module registration — nothing else needed in `getPackages()`.

**5. Android — `MainActivity.kt`**

Extend `SalesforceReactActivity` instead of `ReactActivity`:
```kotlin
class MainActivity : SalesforceReactActivity() {
    override fun getMainComponentName() = "YourApp"
}
```

That's the core integration. The SDK handles OAuth, token refresh, and re-authentication automatically.

**Further reading:**
- Architecture overview: [`docs/ARCHITECTURE.md`](https://github.com/forcedotcom/SalesforceMobileSDK-ReactNative/tree/dev/docs/ARCHITECTURE.md)
- JavaScript API reference: [`docs/javascript/API_REFERENCE.md`](https://github.com/forcedotcom/SalesforceMobileSDK-ReactNative/tree/dev/docs/javascript/API_REFERENCE.md)
- Push notifications: [`docs/push/README.md`](https://github.com/forcedotcom/SalesforceMobileSDK-ReactNative/tree/dev/docs/push)

The best reference for a complete working example remains the generated template — running `forcereact createwithtemplate` and comparing with your existing app. If you hit a specific integration issue, please open a new issue with details and we'll help troubleshoot.
